### ISSUE TITLE
Creating performance benchmark suite for collision detection

### ISSUE THREAD

**Issue by spydon** [flame-engine/flame#2628]
# What could be improved
It would be great to have a benchmark harness for the collision detection system. 

# Why should this be improved
So that it is easier to know if you're making improvements or regressions, and also a good test bed to try out new broadphase algorithms.

# Any risks?
No change to the library code, so no risk.

# More information
This could possibly be used:
https://pub.dev/packages/benchmark_harness

**Comment by TheMaverickProgrammer** [comment]
What work needs to be done for this? Collision detection is a skillset of mine. I'm interested.

**Comment by TheMaverickProgrammer** [comment]
> @TheMaverickProgrammer we'd like to have a benchmarking suite based on [benchmark_harness](https://pub.dev/packages/benchmark_harness), it should compare different collision detection scenarios and compare different broadphase algorithms.
> 
> If you're interested in collision detection, I'm sure there are some interesting new broadphase algorithms that could be implemented.

Yes there are a few. All of varying degrees of improvement depending on the environment and ratio of static-to-dynamic bodies. This is an area of research I've been nosing around in so it seems like the perfect place for me to contribute directly.
